kie-server-parent/kie-server-api/src/test/java/org/kie/server/api/model/DroolsCommandMarshallingTest.java

```diff
@@ -0,0 +1,175 @@
+                                                      "      \"out-identifier\" : \"fact-pojo1\",\n" +
+                                                      "      \"return-object\" : true,\n" +
+                                                      "      \"entry-point\" : \"DEFAULT\",\n" +
+                                                      "      \"disconnected\" : false\n" +
+                                                      "    }\n" +
+                                                      "  }, {\n" +
+                                                      "    \"fire-all-rules\" : {\n" +
+                                                      "      \"max\" : -1,\n" +
+                                                      "      \"out-identifier\" : \"fire-result\"\n" +
+                                                      "    }\n" +
+                                                      "  } ]\n" +
+                                                      "}";
+
+    private static BatchExecutionCommand createTestCommand() {
+
+        KieCommands commandsFactory = KieServices.Factory.get().getCommands();
+        List<Command<?>> commands = new ArrayList<Command<?>>();
+        Pojo3 pojo3 = new Pojo3("A");
+        Pojo2 pojo2 = new Pojo2("B", true, pojo3);
+        Pojo1 pojo1 = new Pojo1("C", pojo2);
+        commands.add(commandsFactory.newInsert(pojo1, "fact-pojo1"));
+        commands.add(commandsFactory.newFireAllRules("fire-result"));
+        BatchExecutionCommand batchExecution = commandsFactory.newBatchExecution(commands);
+        return batchExecution;
+    }
+
+    private Set<Class<?>> getCustomClasses() {
+        HashSet<Class<?>> classes = new HashSet<Class<?>>();
+        classes.add(Pojo1.class);
+        classes.add(Pojo2.class);
+        classes.add(Pojo3.class);
+        return classes;
+    }
+
+    @Test
+    public void testJaxb() {
+        Marshaller marshaller = MarshallerFactory.getMarshaller(getCustomClasses(), MarshallingFormat.JAXB, getClass().getClassLoader());
+        verifyMarshallingRoundTrip(marshaller, createTestCommand(), jaxbExpectedPayload);
+    }
+
+    @Test
+    public void testXStream() {
+        Marshaller marshaller = MarshallerFactory.getMarshaller(getCustomClasses(), MarshallingFormat.XSTREAM, getClass().getClassLoader());
+        verifyMarshallingRoundTrip(marshaller, createTestCommand(), xstreamExpectedPayload);
+    }
+
+    @Test
+    public void testJSON() {
+        Marshaller marshaller = MarshallerFactory.getMarshaller(getCustomClasses(), MarshallingFormat.JSON, getClass().getClassLoader());
+        verifyMarshallingRoundTrip(marshaller, createTestCommand(), jsonExpectedPayload);
+    }
+
+    private void verifyMarshallingRoundTrip(Marshaller marshaller, BatchExecutionCommand inputCommand, String expectedPayload) {
+        String rawContent = marshaller.marshall(inputCommand);
+        logger.info(rawContent);
```

Comment:
Can you please change the log level to debug?

Reply:
@sutaakar Sure, I'll do it in another JIRA.

Reply:
Filed https://issues.redhat.com/browse/DROOLS-6577